src/components/user/PersonalDetails.vue

```diff
@@ -0,0 +1,344 @@
+const newPasswordTouched = ref(false);
+const newPasswordInputType = ref("password");
+const birthdayTouched = ref(false);
+
+const togglePasswordInputType = () => {
+  return passwordInputType.value === "password"
+    ? (passwordInputType.value = "text")
+    : (passwordInputType.value = "password");
+};
+
+const toggleNewPasswordInputType = () => {
+  return newPasswordInputType.value === "password"
+    ? (newPasswordInputType.value = "text")
+    : (newPasswordInputType.value = "password");
+};
+
+const toggleNewPassword = () => {
+  return (showNewPassword.value = !showNewPassword.value);
+};
+
+const emailError = () => {
+  if (!updatedUser.value.email) return "Please enter your email";
+  if (!regex.email.test(updatedUser.value.email))
+    return "Please enter correct email";
+  return "";
+};
+
+const passwordError = () => {
+  if (!updatedUser.value.password && !passwordTouched.value) return "";
+  if (!updatedUser.value.password && passwordTouched.value)
+    return "Please enter your password";
+  if (updatedUser.value.password.length < 8 && passwordTouched.value)
+    return "At least 8 characters";
+  return "";
+};
+
+const newPasswordError = () => {
+  if (!updatedUser.value.newPassword && !newPasswordTouched.value) return "";
+  if (!updatedUser.value.newPassword && newPasswordTouched.value)
+    return "Please enter your password";
+  if (
+    updatedUser.value.newPassword.length < 8 ||
+    !regex.oneLetter.test(updatedUser.value.newPassword) ||
+    !regex.oneDigit.test(updatedUser.value.newPassword)
+  )
+    return "At least 8 characters, one letter and one digit";
+  return "";
+};
+
+const firstNameError = () => {
+  if (!updatedUser.value.firstName) return "Please enter your first name";
+  if (!regex.name.test(updatedUser.value.firstName))
+    return "Please enter correct first name";
+  return "";
+};
+
+const lastNameError = () => {
+  if (!updatedUser.value.lastName) return "Please enter your last name";
+  if (!regex.name.test(updatedUser.value.lastName))
+    return "Please enter correct last name";
+  return "";
+};
+
+const birthdayError = () => {
+  if (!updatedUser.value.birthday) return "error";
+  if (!isAdult(updatedUser.value.birthday) && birthdayTouched.value)
+    return "error";
+  if (isAdult(updatedUser.value.birthday) && birthdayTouched.value)
+    return "correct";
+  return "";
+};
+
+const isFormValid = () => {
+  let isValid = false;
+  if (
+    updatedUser.value.email &&
+    !emailError() &&
+    updatedUser.value.password &&
+    !passwordError() &&
+    updatedUser.value.firstName &&
+    !firstNameError() &&
+    updatedUser.value.lastName &&
+    !lastNameError() &&
+    updatedUser.value.birthday
+  )
+    isValid = true;
+  if (birthdayTouched.value && birthdayError() === "error") isValid = false;
+  if (
+    (showNewPassword.value && !updatedUser.value.newPassword) ||
+    (showNewPassword.value && newPasswordError())
+  )
+    isValid = false;
+  return isValid;
+};
```

Comment:
Shouldn't all of these just be computed properties? This way we do not need to call them and always have their value up to date

Reply:
At some point I had some problem with computed and decided to go with functions, but now I corrected and everything works fine :)